### ISSUE TITLE
Change Request: Support exported annotation for labeled statements

### ISSUE THREAD

**Issue by xM8WVqaG** [eslint/eslint#17112]
### ESLint version

8.39.0

### What problem do you want to solve?

I upgraded a very old bit of JavaScript to a newer eslint version and found my ancient `.eslintrc` config (pre-dating eslint file extension support) now raised an error:

```js
foo: {
    bar();
}
```

```text
   6:1   error  'foo:' is defined but never used  no-unused-labels
```

Apparently the ancient config used to work around this by defining it as a `global` in a previous version (??):

```json
{
  "globals": {
    "foo": true
  }
}
```

But this no longer works, neither does setting the following inline configuration:

```js
/* exported foo */
```

It seems there is no way to mark up a labeled statement as used elsewhere. I would prefer that than completely disabling `no-unused-labels` because this codebase is one of the few places that rule might actually catch issues!

### What do you think is the correct solution?

[`no-unused-vars`](https://eslint.org/docs/latest/rules/no-unused-vars) lets you declare your intention with `/* exported foo */`.

`no-unused-vars` also has a couple of other configuration options, like being able to ignore `globals` which would be grand:

The below example is adapted from the `no-unused-vars` documentation here: <https://eslint.org/docs/latest/rules/no-unused-vars#vars-local>.

```json
{
  "globals": {
    "foo": true
  },
  "rules": {
    "no-unused-labels": ["error", { "labels": "local" }]
  }
}
```

It would be convenient if [`no-unused-labels`](https://eslint.org/docs/latest/rules/no-unused-labels) supported similar functionality.

I expect this isn't a common issue as labeled statements are not very fashionable.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Hi @xM8WVqaG, thanks for the issue!

> It seems there is no way to mark up a labeled statement as used elsewhere.

How can a label be used elsewhere?

**Comment by xM8WVqaG** [comment]
I'm so sorry to waste your time! I've realised this is an invalid request.

I just tried to create a MWE and I couldn't get this usecase to work in current versions of NodeJS or Firefox.

Some background on the code: it has an outer loop defined in one file, which calls functions defined in other files that in turn reference the original label with the usual `continue foo`, etc syntax. When I opened this issue I knew of labelled statements but had never written them myself so assumed that was valid syntax. I now understand it isn't!

i did some more investigation to find out how this ever worked and I discovered a previous developer had written a tool to _inline those functions as part of the build process_ to turn this back into valid JavaScript.

I'm sorry again for bothering you with this issue, and also to anybody who was offended by reading about the build tooling.
